index.js

```diff
@@ -21,20 +21,19 @@ var PGPubsub = function (conString, options) {
   this.conFails = 0;
 
   options = options || {};
+  options.retries = options.retries || 9;
 
   this.retry = new Retry({
     name: 'pubsub',
     try: function () {
       var db = new pg.Client(self.conString);
-      db.on('error', function () {
-        self.retry.reset();
-        if (self.channels.length) {
-          self.retry.try();
-        }
-      });
       return new Promise(function (resolve, reject) {
+        db.on('error', reject);
```

Comment:
@voxpelli Okay, I see. I'll take this into consideration and adjust the patch accordingly. 

We're currently testing this change internally. When I change anything, I'll `push -f` into this PR.

Thanks for the feedback! :)